src/core/scalebar/qgsticksscalebarrenderer.cpp

```diff
@@ -114,13 +115,44 @@ void QgsTicksScaleBarRenderer::draw( QgsRenderContext &context, const QgsScaleBa
   // we render the bar symbol-layer-by-symbol-layer, to avoid ugliness where the lines overlap in multi-layer symbols
   for ( int layer = 0; layer < symbol->symbolLayerCount(); ++ layer )
   {
-    // first draw the vertical lines
+    // first draw the vertical lines for segments
     for ( int i = 0; i < positions.size(); ++i )
     {
       const double thisX = context.convertToPainterUnits( positions.at( i ), QgsUnitTypes::RenderMillimeters ) + xOffset;
       symbol->renderPolyline( QPolygonF() << QPointF( thisX, barTopPosition )
                               << QPointF( thisX, bottomPosition ), nullptr, context, layer );
     }
+    // vertical positions
+    double verticalPos = 0.0;
+    QList<double> subTickPositionsY;
+    switch ( mTickPosition )
+    {
+      case TicksDown:
+        verticalPos = barTopPosition;
+        subTickPositionsY << barTopPosition;
+        subTickPositionsY << middlePosition;
+        break;
+      case TicksMiddle:
+        verticalPos = middlePosition;
+        subTickPositionsY << (barTopPosition + middlePosition ) / 2;
```

Comment:
I think ideally we'd expose the subdivision height to user control

Reply:
Hi @nyalldawson, thanks for the revision of this preliminary work.
What do you think about letting the user choose the subdivisions tick height <= the segments tick height?

Reply:
I think it's a good idea. We could either expose it as an absolute mm size or a percentage (of tick height) size...

Reply:
Subdivisions tick height control added.